homeassistant/components/climate/radiotherm.py

```diff
@@ -77,19 +111,25 @@ class RadioThermostat(ClimateDevice):
     def __init__(self, device, hold_temp, away_temps):
         """Initialize the thermostat."""
         self.device = device
+        # It would be better if this was in update() since it triggers
+        # a network call but the thermostat will clear any temporary
+        # mode or temperature if this is called so we have to leave it here.
         self.set_time()
         self._target_temperature = None
         self._current_temperature = None
         self._current_operation = STATE_IDLE
         self._name = None
         self._fmode = None
+        self._fstate = None
         self._tmode = None
         self._tstate = None
         self._hold_temp = hold_temp
+        self._hold_set = False
         self._away = False
         self._away_temps = away_temps
         self._prev_temp = None
         self._operation_list = [STATE_AUTO, STATE_COOL, STATE_HEAT, STATE_OFF]
+        self._fan_list = [STATE_ON, STATE_AUTO]
```

Comment:
Same as above.

Reply:
It seems like the device can report three fan modes, but here the user can only select two modes. Is that correct?

Reply:
Thanks - I went back and checked the docs more carefully.  The circulate mode is only supported by one model so I changed it to detect that model and use two different operations list depending on that flag.